src/tests/Common/tests.targets

```diff
@@ -69,7 +69,10 @@
 
       <CorerunExecutable Condition="'$(RunningOnUnix)' == 'true'"  >$(CORE_ROOT)\corerun</CorerunExecutable>
       <CorerunExecutable Condition="'$(RunningOnUnix)' != 'true'">$(CORE_ROOT)\corerun.exe</CorerunExecutable>
-      <CorerunExecutable Condition="'$(TargetOS)' == 'Browser' Or '$(TargetOS)' == 'Android' or '$(TargetOS)' == 'iOSSimulator'"> $(DotnetRoot)/dotnet</CorerunExecutable>
+      <DotnetExecutable>dotnet --roll-forward LatestMajor</DotnetExecutable>
```

Comment:
Why is this needed?

Reply:
Getting

```
  dotnet C:\git\runtime2\artifacts\tests\coreclr\windows.x64.Debug\Tests\Core_Root\xunit\xunit.console.dll .\nativeaot\SmokeTests\nativeaot.SmokeTests.XUnitWrapper.dll -parallel collections -html C:\git\runtime2\artifacts\log\TestRun.html -xml C:\git\runtime2\artifacts\log\TestRun.xml  -notrait category=outerloop -notrait category=failing -nocolor
  Microsoft.DotNet.XUnitConsoleRunner v2.5.0 (64-bit .NET Core 3.1.21)
     nativeaot.SmokeTests.XUnitWrapper: Exception thrown during test framework discoverer construction: System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
```

locally without that. Not sure why it's choosing a 3.1 version of the runtime to run xunit. The host resolution is a mystery.

Reply:
It looks like it's trying to pick the host to match `xunit.console.dll` since there's no `-runtimeconfig` parameter to point to a different `runtimeconfig.json`. Since `xunit.console.dll` is probably built against .NET Core 3.1, that's what it selects for the runtime without `LatestMajor` roll-forward.

Reply:
Thank you @jkoritzinsky! Any preference for solving this or is roll forward okay?

Alternatively, I could change these legs to build full CoreCLR and then corerun is available and this change is not needed. But felt odd having to build entire CoreCLR just to run xunit. I guess we could also AOT compile the xunit runner, but that sounds like a lot of additional build complexity.

Reply:
I’m fine keeping the roll-forward for now as we can/should probably use the new xunit wrapper generator instead of the console runner if we want to AOT this.

Can you add a comment here though about the reason for the roll-forward so we don’t forget?